modin/core/io/file_dispatcher.py

```diff
@@ -276,14 +278,21 @@ def file_exists(cls, file_path, storage_options=None):
                 else:
                     new_storage_options = {}
 
-                s3fs = S3FS.S3FileSystem(anon=False, **new_storage_options)
+                fs, _ = fsspec.core.url_to_fs(file_path, **new_storage_options)
                 exists = False
                 try:
-                    exists = s3fs.exists(file_path) or exists
-                except (NoCredentialsError, PermissionError, EndpointConnectionError):
+                    exists = fs.exists(file_path)
+                except (
+                    NoCredentialsError,
+                    PermissionError,
+                    EndpointConnectionError,
+                    ConnectTimeoutError,
+                ):
                     pass
-                s3fs = S3FS.S3FileSystem(anon=True, **new_storage_options)
-                return exists or s3fs.exists(file_path)
+                fs, _ = fsspec.core.url_to_fs(
+                    file_path, anon=True, **new_storage_options
+                )
```

Comment:
I would rather we check for `anon=True` only in `except:` block - there's no need to try an anonymous access if normal one worked fine

Reply:
Moved to `except` block